src/core_ocean/shared/mpas_ocn_vel_tidal_potential.F

```diff
@@ -0,0 +1,210 @@
+      real (kind=RKIND) :: invdcEdge 
+      real (kind=RKIND) :: potentialGrad 
+      logical, pointer :: config_use_tidal_potential_forcing
+
+      err = 0
+
+      if (.not. tidalPotentialOn) return
+
+      call mpas_pool_get_dimension(meshPool, 'nEdgesArray', nEdgesArray)
+      call mpas_pool_get_array(meshPool, 'maxLevelEdgeTop', maxLevelEdgeTop)
+      call mpas_pool_get_array(meshPool, 'cellsOnEdge', cellsOnEdge)
+      call mpas_pool_get_array(meshPool, 'dcEdge', dcEdge)
+      call mpas_pool_get_array(meshPool, 'edgeMask', edgeMask)
+      call mpas_pool_get_array(forcingPool, 'tidalPotentialEta', tidalPotentialEta)
+
+      call mpas_pool_get_config(ocnConfigs, 'config_use_tidal_potential_forcing', config_use_tidal_potential_forcing)
+      call mpas_pool_get_config(ocnConfigs, 'config_self_attraction_and_loading_beta', config_self_attraction_and_loading_beta)
+
+      nEdges = nEdgesArray( 1 )
+
+      !$omp do schedule(runtime) private(cell1, cell2, invdcEdge, potentialGrad, k)
+      do iEdge=1,nEdges
+         cell1 = cellsOnEdge(1,iEdge)
+         cell2 = cellsOnEdge(2,iEdge)
+         invdcEdge = 1.0_RKIND / dcEdge(iEdge)
+
+         potentialGrad = gravity * invdcEdge * ( tidalPotentialEta(cell2) - tidalPotentialEta(cell1) &
+                                 + config_self_attraction_and_loading_beta*(ssh(cell2) - ssh(cell1)))
+
+         do k=1,maxLevelEdgeTop(iEdge)
+            tend(k,iEdge) = tend(k,iEdge) -  edgeMask(k,iEdge) * potentialGrad 
+         end do
+      end do
+      !$omp end do
+
+   !--------------------------------------------------------------------
+
+   end subroutine ocn_vel_tidal_potential_tend!}}}
+
+!***********************************************************************
+!
+!  routine ocn_vel_tidal_potential_init
+!
+!> \brief   Initializes ocean momentum tidal potential gradient
+!> \author  Steven Brus 
+!> \date    April 2020 
+!> \details
+!>  This routine initializes parameters required for the computation of the
+!>  tidal potential gradient.
+!
+!-----------------------------------------------------------------------
+
+   subroutine ocn_vel_tidal_potential_init(err)!{{{
+
+   !--------------------------------------------------------------------
+
+
+      !-----------------------------------------------------------------
+      !
+      ! Output Variables
+      !
+      !-----------------------------------------------------------------
+
+      integer, intent(out) :: err !< Output: error flag
+
+
+      !-----------------------------------------------------------------
+      !
+      ! call individual init routines for each parameterization
+      !
+      !-----------------------------------------------------------------
+      logical, pointer :: config_use_tidal_potential_forcing
+
+      err = 0
+
+      call mpas_pool_get_config(ocnConfigs, 'config_use_tidal_potential_forcing', config_use_tidal_potential_forcing)
+
+      tidalPotentialOn= .true.
```

Comment:
Should this be initialized to false?  Obviously having it set to true helps for testing but in the final version it seems like it should be false.

Reply:
Yes, absolutely.